### PR TITLE
Adding better doc-strings for retrieval & ranking models

### PR DESCRIPTION
**Pull request by marcromeyn**
Fixes # (issue)

### CODE REVIEW

**Comment by benfred** [comment]
@marcromeyn can we get this updated for the release on tuesday?

File: merlin/models/tf/models/utils.py [hunk 1]
```diff
@@ -1,19 +0,0 @@
-from typing import List, Optional, Union
```
Comment (benfred) [inline merlin/models/tf/models/utils.py:1]: I think we need to adjust the RST stubs in docs/source/api to reflect that this file is deleted (its the reason its failing CI now)

**Comment by viswa-nvidia** [comment]
Moved to arbitration. Not clear how this maps to the r[oadmap](https://github.com/orgs/NVIDIA-Merlin/projects/14/views/4) . @marcromeyn , please clarify.

**Comment by viswa-nvidia** [comment]
Moved to arbitration. Not clear how this maps to the r[oadmap](https://github.com/orgs/NVIDIA-Merlin/projects/14/views/4) . @marcromeyn , please clarify.

**Comment by bschifferer** [comment]
@marcromeyn is this PR still relevant? Are you active working on it? Can we close it OR move it to need prioritization/clarification?

**Comment by viswa-nvidia** [comment]
We have a new API so this one can be closed

**Pull request closed without merging**